File: backend/app/schemas/transaction.py
from datetime import datetime
from pydantic import BaseModel
import enum


class TransactionType(str, enum.Enum):
    BUY = "BUY"
    SELL = "SELL"
    ADJUST = "ADJUST"
    DEPOSIT = "DEPOSIT"
    WITHDRAW = "WITHDRAW"


class TransactionCreate(BaseModel):
    holding_id: int | None = None
    type: TransactionType
    quantity: float
    price: float
    currency: str | None = None
    notes: str | None = None
    transacted_at: datetime


class TransactionResponse(BaseModel):
    id: int
    holding_id: int | None
    type: str
    quantity: float
    price: float
    total_amount: float
    currency: str | None = None
    notes: str | None
    transacted_at: datetime
    created_at: datetime
    holding_name: str | None = None
    holding_symbol: str | None = None

    model_config = {"from_attributes": True}

File: frontend/src/types/transaction.ts
export interface Transaction {
  id: number
  holding_id: number
  type: 'BUY' | 'SELL' | 'ADJUST'
  quantity: number
  price: number
  total_amount: number
  notes: string | null
  transacted_at: string
  created_at: string
  holding_name: string | null
  holding_symbol: string | null
}

export interface TransactionCreate {
  holding_id: number
  type: 'BUY' | 'SELL' | 'ADJUST'
  quantity: number
  price: number
  notes?: string
  transacted_at: string
}

export interface TransactionUpdate {
  transacted_at?: string
  notes?: string
}
